samples/java-customer-registry/src/main/java/customer/CustomerValueEntityProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package customer;
+
+import com.akkaserverless.javasdk.valueentity.ValueEntityContext;
+import com.akkaserverless.javasdk.valueentity.ValueEntityProvider;
+import com.google.protobuf.Descriptors;
+import customer.api.CustomerApi;
+
+import java.util.function.Function;
+
+// editable, to be generated in generated-sources/src/main/java
+public class CustomerValueEntityProvider implements ValueEntityProvider<CustomerValueEntity> {
```

Comment:
I'm not so sure how we should call this. 

It's not really a ValueEntityProvider. Instead, it provides the handler and some other required configurations (descriptors and type hint).

Reply:
I think ValueEntityProvider is fine: from the users' perspective this is the thing that produces value entity instances, and that this is a 'multiple-step process' with a handler in between is an implementation detail.